exec/cnex/array.c

```diff
@@ -36,6 +36,29 @@ Array *array_createArrayFromSize(size_t iElements)
     return a;
 }
 
+Array *array_expandArray(Array *self, size_t iElements)
+{
+    if (self == NULL) {
+        self = malloc(sizeof(Array));
+        if (self == NULL) {
+            fatal_error("Unable to allocate array for array expansion.");
+        }
+        self->size = 0;
+        self->data = NULL;
+    }
+
+    self->data = realloc(self->data, sizeof(Cell) * self->size + iElements);
```

Comment:
The size parameter passed to realloc doesn't look right. Check your order of operations.

Also, Is this function intended to *add* iElements new elements to an existing array, or is it intended to *set* the number of elements to iElements (including shrinking)? I can't really be sure from this code and how it's called.

Reply:
This is meant to expand the array BY iElements.  Not to make the array size BE iElements when it's all over.  Corrected the math on this call.